go/processor/floria/run_context.go

```diff
@@ -76,9 +76,11 @@ func (r runContext) Call(kind tosca.CallKind, parameters tosca.CallParameters) (
 	}
 
 	snapshot := r.CreateSnapshot()
-	if err := transferValue(r, parameters.Value, parameters.Sender, recipient); err != nil {
-		r.RestoreSnapshot(snapshot)
-		return tosca.CallResult{}, nil
+	if kind != tosca.DelegateCall && kind != tosca.CallCode {
```

Comment:
I would cover such features with unit tests to specifically target the changed feature. 

Integration tests are good for comparing between implementations and for smoke testing, but identified requirements like "the transfer needs to fail if the sender has not enough funds, even if it is the receiver" or "no value transfer for delegate calls" can be tested more to the point using unit tests.

It would also document these requirements.